Overhauls the auto-transfer vote system

Rukofamicom opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

This replaces the auto-transfer vote that is easily missed with a persistent system that allows players to vote to leave at any time they wish instead of waiting for the next available voting time, with votes being checked every five minutes while also respecting the shuttle refueling time that prevents a shuttle from being called for the first 20 minutes right now. 

This system features three configuration variables:
* The initial percentage of votes required to call the shuttle, (default 75%)
* How much time should pass before this percentage starts to decay, also used to determine how often players are reminded that the voting system exists to encourage players to consider whether they believe the round should end.  (default 60 minutes)
* How quickly the required percentage of votes decays every five minutes, after the decay first starts (default 2.5%)

What this breaks down to mean, if the given defaults are used:
* The shuttle cannot be called until at least 20 minutes have passed regardless of how many players vote, due to the currently configured refueling time before shuttles may be called. 
* Players must overwhelmingly vote to call an early shuttle (first hour of round) with 75% of players being required to call a shuttle between 20 and 60 minutes
* After 60 minutes the required percentage of players gradually decays every voting check
* At 1:30 the vote requires 60% to call a shuttle
* At 2:00 the vote requires has started requiring less than half of players at only 45% required to call
* At 2:30 the vote requires only 30% to call
* By 3:30 the shuttle is forcibly called because rounds should never really exceed this time limit, even with a unanimous vote to leave just to ensure unforseen shenanigans can't happen.

**Coincidentally this new voting system does not care if players are alive or dead.**

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

The current voting system is functional but somewhat flawed for a few reasons.
* **Only living players get to vote.** Usually this is a good thing in my opinion, but every now and then it results in the round being held entirely hostage, especially at lower populations. 
* **After a certain time limit, either admins or dynamic takes over**. There is very rarely a natural end to rounds, it feels like the only way rounds end under the current system is by shit absolutely spraying through a fan most of the time. This might be a good thing to some people, but not to all of them and I think this voting system will enable a better balance of chaos and order when it comes to round endings.

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding.
-->

## Testing Photographs and Procedure
<!-- Include any screenshots/videos/debugging steps of the modified code functioning successfully, ideally including edge cases. -->

My exact testing procedure involved debugging a lot of variables, This is the most succinct screenshot I can provide to show that the system is working as intended. 

![image](https://github.com/user-attachments/assets/9dae2bcd-a8e4-47de-be78-a01c98e69c1e)

![image](https://github.com/user-attachments/assets/a1afda85-ee15-4b9a-92b9-0ebaee71e5a8)


## Changelog
:cl:
add: Added a new voting system. Players will find a new button in their status tab for voting to end the current round which they can use at any time instead of needing to wait. The required percentage of players for the vote to succeed decays as the round goes on, and the button first becomes visible only after the shuttle is available. 
tweak: Dead players can now vote to end a round
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Rukofamicom committed:
Overhauls the auto-transfer system

Rukofamicom committed:
remove unnecessary config

Rukofamicom committed:
updates the reminder tip

EvilDragonfiend commented on code/_globalvars/lists/client.dm:
```diff
@@ -1,2 +1,3 @@
 GLOBAL_LIST_EMPTY(keybindings_by_name)
 GLOBAL_LIST_EMPTY(keybindings_by_name_to_key)
+GLOBAL_LIST_EMPTY(total_votes_to_leave)
```
This GLOB should be in SS.

Rukofamicom replied:
There isn't an SS file for globals.

Rukofamicom replied:
Alright, I made a subsystem.dm file just for this.

EvilDragonfiend replied:
Bacon resolved this now.

EvilDragonfiend reviewed (changes requested):
I am not sure about this, but trying out is nothing bad anyways.

Rukofamicom committed:
code improvements per ike

Rukofamicom committed:
creates a new subsystem file just for this one line

Rukofamicom committed:
Update client.dm

PowerfulBacon committed:
Auto transfer stat panel

PowerfulBacon committed:
Update mob_stat.dm

Rukofamicom committed:
Merge pull request #15 from PowerfulBacon/auto-transfer-stat-panel

Auto transfer stat panel

PowerfulBacon commented on code/modules/mob/mob_stat.dm:
```diff
@@ -245,6 +245,11 @@
 		if(ETA)
 			tab_data[ETA] = GENERATE_STAT_TEXT(SSshuttle.emergency.getTimerStr())
+	if (!isnewplayer(src) && SSautotransfer.can_fire)
+		if (SSautotransfer.required_votes_to_leave && SSshuttle.canEvac())
```
SSshuttle.canEvac returns a string, which means its always true. At the very least it shouldn't show the players voting to leave until the shuttle is ready to be called (as early in the round you could meta the number of people). The more restrictions the better, but that one is my minimum.

Rukofamicom replied:
fixed with the same modification I gave in my original code and also almost the same comment I had next to it warning that `canEvac()` always returns a value. 

Read the comments!

PowerfulBacon reviewed (commented):
Other than that, I really like this system and I think it works a lot better than the current system for crew transfer.

Rukofamicom committed:
Update mob_stat.dm

updates the canEvac() check so it works again. 
Also replaces the comment warning how it works which was deleted when this was modified.

PowerfulBacon commented on code/controllers/subsystem/autotransfer.dm:
```diff
@@ -3,19 +3,49 @@ SUBSYSTEM_DEF(autotransfer)
+			connected_votes_to_leave ++
+
+	if(REALTIMEOFDAY > checkvotes_time)
+		if(decay_start)
+			decay_count++
```
This should be calculated from the time rather than through a maintained state to make it less prone to errors and lag.

Rukofamicom replied:
tbh it is calculated through time since this can only fire once every five minutes.

Rukofamicom committed:
Spanless

Rukofamicom committed:
MISSING INDENTATION AAAAAA

PowerfulBacon reviewed (changes requested):
I'm happy with this, but I want the config values written into the config.txt file that the codebase uses like the other config entries.

Rukofamicom committed:
Updates config

PowerfulBacon reviewed (approved):
Appears to be functioning as intended and in a sufficient manner. Config needs to be applied properly on the server prior to full merge if it hasn't been already.

PowerfulBacon reviewed (approved):
Config updated
